widgets/heatmap/heatmap.go

```diff
@@ -76,7 +88,50 @@ func New(opts ...Option) (*HeatMap, error) {
 // Each call to Values overwrites any previously provided values.
 // Provided options override values set when New() was called.
 func (hp *HeatMap) Values(xLabels []string, yLabels []string, values [][]float64, opts ...Option) error {
-	return errors.New("not implemented")
+	if len(values) == 0 {
+		return errors.New("the values cannot be empty")
+	}
+
+	hp.mu.Lock()
+	defer hp.mu.Unlock()
+
+	var (
+		xl []string
```

Comment:
Since these variables have larger scope (the full function), the we give them longer (more readable) names?

Reply:
I renamed `v` to `vals`, but don't know how to deal with `xl` and `yl`. `xLabels` and `yLabels` have been used. Do you have any idea ?